### PR TITLE
Swift write from provider

### PR DESCRIPTION
**Pull request by ashleylst**
Add uploading files function for Swift, with multipart upload considered. 

One can use 'davix-put local/file/path swifts://.. --ostoken $token --osprojectid $id' to upload local file to Swift, and use '-r' option to upload a directory of files to Swift. One can use fragment '#forceMultipart' to make use of [multipart upload](https://docs.openstack.org/swift/latest/overview_large_objects.html#module-swift.common.middleware.slo) supported by Swift.

Multipart upload is used for files > 512MB. In this case, a large file is uploaded in a set of segments of maximum size 256MB. The default maximum segment number in one multipart manifest in Swift is 1000, hence, large files that exceed the 1000 limit in segments will be uploaded in an inline manner (where the file is represented by manifests within one manifest).

### CODE REVIEW

File: src/fileops/SwiftIO.cpp [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+            manifest << "\"size_bytes\":" << props[i - 1].second << "}";
+            if(i != propEnd) {
+                manifest << ',';
+            }
+        }
+        manifest << "]";
+
+        remaining -= propEnd - propBegin + 1;
+        propBegin = propEnd + 1;
+        propEnd = std::min(propEnd + MaxManifestSegments - 1, propEnd + remaining);
```
Comment (mpatrascoiu) [inline src/fileops/SwiftIO.cpp:148]: Shouldn't it be `propEnd + MaxManifestSegments`? 
(as you're increasing it by exactly one interval  `= MaxManifestSegments`)

**Review by mpatrascoiu — COMMENTED** [review]
I found the calculations in `SwiftIO::commitInlineChunks` a bit hard to follow, namely the `propEnd` because it's tied to `remaining` variable, when it doesn't have to be.

There's `propBegin`, which always increases.
There's `propEnd`, which increases by `MaxManifestsSegments`, with the additional (final) case where it increases only by the remainder. Max size is `props.size()`.

But then there's `remaining` variable, which is decreasing.

Wouldn't using just `propEnd` work?
```cpp
propEnd = std::min(propEnd + MaxMinfestSegments, props.size());
```
However, this would change your terminal condition to
```cpp
propBegin > propEnd
```
which is not as explicit as `remaining == 0`.
Food for thought.

That was my only comment, rest looks very good.
Many thanks!

**Comment by ashleylst** [comment]
Thanks for the comment, I did not realize I was using an additional variable, as I focused on seeing if it is possible to commit chunks in an inline manner. Now it should be fine.
